src/Models/ContainerDeviceBase.ts

```diff
@@ -77,97 +77,221 @@ export abstract class ContainerDeviceBase implements Device {
-    const scaffoldType = ScaffoldType.Local;
-    if (!await FileUtility.directoryExists(scaffoldType, this.projectFolder)) {
+    const createTimeScaffoldType = ScaffoldType.Local;
+    if (!await FileUtility.directoryExists(
+            createTimeScaffoldType, this.projectFolder)) {
       throw new Error('Unable to find the project folder.');
     }
 
-    await this.generateTemplateFiles(scaffoldType, this.templateFilesInfo);
+    await this.generateTemplateFiles(
+        createTimeScaffoldType, this.projectFolder, this.templateFilesInfo);
 
-    return true;
+    const res = await this.configDeviceEnvironment(
+        this.projectFolder, createTimeScaffoldType);
+
+    return res;
   }
 
   async generateTemplateFiles(
-      type: ScaffoldType,
+      type: ScaffoldType, projectPath: string,
       templateFilesInfo: TemplateFileInfo[]): Promise<boolean> {
     if (!templateFilesInfo) {
       throw new Error('No template file provided.');
     }
 
+    if (!projectPath) {
+      throw new Error(`Project path is empty.`);
+    }
+
     // Cannot use forEach here since it's async
     for (const fileInfo of templateFilesInfo) {
-      await generateTemplateFile(this.projectFolder, type, fileInfo);
+      // Replace binary name in CMakeLists.txt to project name
+      if (fileInfo.fileName === 'CMakeLists.txt') {
+        const pattern = '${project_name}';
+        const projectName = path.basename(projectPath);
+        if (fileInfo.fileContent) {
+          fileInfo.fileContent =
+              fileInfo.fileContent.replace(pattern, projectName);
+        }
+      }
+      await utils.generateTemplateFile(this.projectFolder, type, fileInfo);
     }
     return true;
   }
 
   async compile(): Promise<boolean> {
+    // Check remote
     const isRemote = RemoteExtension.isRemote(this.extensionContext);
     if (!isRemote) {
-      const res = await askAndOpenInRemote(
+      const res = await utils.askAndOpenInRemote(
           OperationType.Compile, this.channel, this.telemetryContext);
       if (!res) {
         return false;
       }
     }
 
+    const compileTimeScaffoldType = ScaffoldType.Workspace;
     if (!await FileUtility.directoryExists(
-            ScaffoldType.Workspace, this.outputPath)) {
-      try {
-        await FileUtility.mkdirRecursively(
-            ScaffoldType.Workspace, this.outputPath);
-      } catch (error) {
-        throw new Error(`Failed to create output path ${
-            this.outputPath}. Error message: ${error.message}`);
-      }
+            compileTimeScaffoldType, this.projectFolder)) {
+      throw new Error('Unable to find the project folder.');
     }
 
+    // Fetch and execute compilation task
     const tasks = await vscode.tasks.fetchTasks();
     if (!tasks || tasks.length < 1) {
+      const message = `Failed to fetch tasks.`;
+      utils.channelShowAndAppendLine(this.channel, message);
+
+      await utils.askToConfigureEnvironment(
```

Comment:
Yes. The function `utils.askToConfigureEnvironmen()` is used to `"Pop out information window suggesting user to configure project environment first."`. 
It is called when user tries to compile / upload device code but fails because `tasks.json` is incomplete. In these situations, user's compile / upload operation has been proved fail, and thus we return false.